src/components/tl/mlx5/mcast/tl_mlx5_mcast_service_coll.c

```diff
@@ -0,0 +1,81 @@
+/**
```

Comment:
There is a lot of verbosity and two new files only for posting service collective. Can't this be avoided?

Reply:
I can think of a few reasons:
1. mcast lib was designed to be as self-contained as possible, making it more portable to other libs. we can transparently move design components from UCC to them.
2. In the mcast internal functions we are not dealing ucc_team/ucc_subset_t, etc. structs, instead, we have a single ucc_tl_mlx5_mcast_oob_p2p_context_t struct that holds everything. therefore, we need a wrapper on top of ucc_service_XX() calls to translate that struct to UCC objects and call ucc_service_XX().
3. Some service calls like Barrier are unavailable as ucc_service_XX() calls, so we implement those. 
4. if in the future, UCC changes its way of service calls, or provides a new API, there would be minimal changes to mcast lib.